src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +407,31 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (method.Equals("$/progress"))
+            {
+                JToken progressId =
+                    args is JObject ? args["token"] :
+                    args is JArray ? args[0] :
+                    null;
+
+                JToken value =
+                    args is JObject ? args["value"] :
+                    args is JArray ? args[1] :
+                    null;
+
```

Comment:
What happens if these expected indexes/names are not present? Presumably we'll throw (sooner or later). But as this is a notification, we should probably just swallow it (and log it to the TraceSource) rather than let it break the connection.